### PR TITLE
CSHARP-3722: Raise client error for snapshot sessions on <5.0 servers

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Operations/ReadConcernHelper.cs [hunk 1]
```diff
@@ -40,6 +40,11 @@ private static BsonDocument ToBsonDocument(ICoreSession session, ConnectionDescr
             // snapshot
             if (sessionsAreSupported && session.IsSnapshot)
             {
+                if (connectionDescription.IsMasterResult.MaxWireVersion < 13)
+                {
+                    throw new MongoClientException("Snapshot reads require MongoDB 5.0 or later");
+                }
+
```
Comment (rstam) [inline src/MongoDB.Driver.Core/Core/Operations/ReadConcernHelper.cs:47]: We use the `Feature` class to determine whether a feature is supported or not.

Replace these lines with:

    Feature.SnapshotReads.ThrowIfNotSupported(connectionDescription.ServerVersion);

(and add `SnapshotReads` to the `Feature` class).

**Review by rstam — CHANGES_REQUESTED** [review]

File: src/MongoDB.Driver.Core/Core/Operations/ReadConcernHelper.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver.Core/Core/Operations/ReadConcernHelper.cs:45]: Done.

**Review by rstam — APPROVED** [review]
LGTM

**Comment by BorisDog** [comment]
Merged [here](https://github.com/BorisDog/mongo-csharp-driver/commit/b947f6e27139b55e58f71cfcf45c2a7b04edeac0).

**Pull request closed without merging**